packages/flutter/lib/src/services/message_codec.dart

```diff
@@ -145,6 +145,51 @@ class PlatformException implements Exception {
   ///   print(stacktrace);
   /// }
   /// ```
+  ///
+  /// On Android this field is populated when an exception is thrown in the method call handler,
```

Comment:
I'm not sure why this needs to be the case, but the exception must also be an unchecked exception such as any subclass of `RuntimeException`.  Those that explicitly throw an exception must populate the stacktrace  https://github.com/flutter/engine/blob/b0c129db124d19e4c55d242701e5792be9d304a8/shell/platform/android/io/flutter/plugin/common/MethodChannel.java#L282-L284.

Reply:
Added in 25d30f7

Reply:
> Those that explicitly throw an exception must populate the stacktrace

That's currently not possible because `Result` doesn't take a stacktrace as an argument to pass it to the `BinaryReply`, see https://github.com/flutter/engine/blob/b0c129db124d19e4c55d242701e5792be9d304a8/shell/platform/android/io/flutter/plugin/common/MethodChannel.java#L271

Reply:
got it. thanks